Changelog.md

```diff
@@ -2,6 +2,17 @@
 
 ## Unreleased
 
+### General
+
+- **Change** For a structure `S` on a subject of type `T`, declares `S.sort` as
+	an Elpi coercion from `S.type` to `T` and `S.pack` as an Epli coercion from
+	`T` to `S.type` whenever `T` is not a global type (e.g. a variable). Note
+	that `S.pack` can cast a `t : T` to `S.type` only if an instance of the
+	class `S` on `t` is found by type class inference
+- **Change** `HB.structure` declares the class of a structure (`axioms_`) as a
+	type class on the subject with all arguments in output mode but for the
+	subject that is in input mode.
+
```

Comment:
Isn't this completely ortogonal to the coercion part?
To me these raises yet unresolved nontrivial. I have nothing against merging it, but, just in case, I would prefer that it is a distinct commit/PR so that we can revert easily if something goes wrong.

Reply:
Sorry for the delay, I rewrote the history as suggested and added a `typeclass` option.